### PR TITLE
Makefile target for shared library build

### PR DESCRIPTION
**Pull request by patlkli**
I added a Makefile target in order to do a very simple shared library build.

If there's any demand, I could also create an autoconf-based solution.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,8 +1,11 @@
-CFLAGS?=-std=c89 -ansi -pedantic -O4 -Wall
+CFLAGS?=-std=c89 -ansi -pedantic -O4 -Wall -fPIC 
 
 default: multipart_parser.o
 
 multipart_parser.o: multipart_parser.c multipart_parser.h
 
+solib: multipart_parser.o
+	$(CC) -shared -Wl,-soname,libmultipart.so -o libmultipart.so multipart_parser.o
```
Comment (iafonov) [inline Makefile]: is it conventional to not have spaces around commas?

**Comment by patlkli** [comment]
Yes, it's GCC's syntax to pass options to the linker.

From the GCC manpage:
```
-Wl,option
           Pass option as an option to the linker.  If option contains commas,
           it is split into multiple options at the commas.  You can use this
           syntax to pass an argument to the option.  For example,
           -Wl,-Map,output.map passes -Map output.map to the linker.  When
           using the GNU linker, you can also get the same effect with
           -Wl,-Map=output.map.
```

**Comment by iafonov** [comment]
:+1:

**Review by albertoblue87-netizen — COMMENTED** [review]
Listo

**Review by albertoblue87-netizen — COMMENTED** [review]
Listo

**Pull request merged**